examples/flutter_gallery/lib/demo/video_demo.dart

```diff
@@ -0,0 +1,361 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:connectivity/connectivity.dart';
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+
+// TODO(sigurdm): These should not be stored here.
+const String butterflyUri =
+    'https://flutter.github.io/assets-for-api-docs/videos/butterfly.mp4';
+
+const String beeUri =
+    'https://flutter.github.io/assets-for-api-docs/videos/bee.mp4';
+
+class VideoCard extends StatelessWidget {
+  final VideoPlayerController controller;
+  final String title;
+  final String subtitle;
+
+  const VideoCard({Key key, this.controller, this.title, this.subtitle})
+      : super(key: key);
+
+  Widget _buildInlineVideo() {
+    return new Padding(
+      padding: const EdgeInsets.symmetric(vertical: 10.0, horizontal: 30.0),
+      child: new Center(
+        child: new AspectRatio(
+          aspectRatio: 3 / 2,
+          child: new Hero(
+            tag: controller,
+            child: new VideoPlayer(controller),
+          ),
+        ),
+      ),
+    );
+  }
+
+  Widget _buildFullScreenVideo() {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text(title),
+      ),
+      body: new Center(
+        child: new AspectRatio(
+          aspectRatio: 3 / 2,
+          child: new Hero(
+            tag: controller,
+            child: new VideoPlayPause(controller),
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Card(
+      child: new Column(
+        children: <Widget>[
+          new ListTile(title: new Text(title), subtitle: new Text(subtitle)),
+          new GestureDetector(
+            onTap: () {
+              final TransitionRoute<Null> route = new PageRouteBuilder<Null>(
+                settings: new RouteSettings(name: title, isInitialRoute: false),
+                pageBuilder: (BuildContext context, Animation<double> animation,
+                    Animation<double> secondaryAnimation) {
```

Comment:
You could avoid this deep nesting by extracting the `onTap` handler and/or the `pageBuilder` as locals above.

Reply:
Done